lib/modules/apostrophe-pages/lib/api.js

```diff
@@ -690,8 +690,60 @@ module.exports = function(self, options) {
   //
   // See `moveToTrash` for what the callback receives.
 
-  self.trashInSchema = function(req, _id, callback) {
-    return self.trashInSchemaBody(req, _id, true, callback);
+  self.trashInSchema = function(req, _id, toTrash, callback) {
+    var page, parent;
+    var children = [];
+
+    return async.series(
+      [
+        function getPage(cb) {
+        // check permissions and load page to trash/untrash
+          return self.find(req, { _id: _id }).permission('edit').trash(null).ancestors({ depth: 1, published: null, trash: null, areas: false }).toObject((err, _page) => {
+            page = _page;
+            parent = page._ancestors[0];
+            if (!page) {
+              return cb('Page not found');
+            }
+            if (!parent) {
+              return cb('Cannot move the home page to or from the trash');
+            }
+            return cb(err);
+          });
+        },
+
+        // get all children of page
+        function getChildren(cb) {
+          const path = self.matchDescendants(page);
+          self.find(req, { path: path }).toArray((err, res) => {
+            if (!err) {
+              children = children.concat(res);
+            }
+            return cb(err, res);
+          });
+        },
+
+        // iterate over pages array: deduplicate requisite fields and flag as trash
+        function dedupeAndTrashPages(cb) {
+          children.push(page);
+          return self.dedupeAndTrashPages(req, children, toTrash, (err, res) => {
+            return cb(err);
+          });
+        }
+      ], err => {
+        return callback(err, parent && parent.slug, children); // @@TODO PAUL check that children is equivalent to changed in the old implementation
+      });
+  };
+
+  self.dedupeAndTrashPages = function(req, pages, toTrash, callback) {
+    return async.forEach(pages, (page, cb) => {
+      if (toTrash) {
+        // @@TOM still a little unclear if I need to grab the manager per page OR
+        // if we are EXTENDING the manager here already -- just to use self.deduplicateTrash
+        return self.apos.docs.getManager(page.type).deduplicateTrash(req, page, cb);
```

Comment:
We're not extending the manager here, so these `getManager` calls are appropriate.